airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -0,0 +1,405 @@
+
+import requests
+import vcr
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.streams.http import HttpStream
+from requests.exceptions import HTTPError
+
+cache_file = tempfile.NamedTemporaryFile()
+
+
+class GithubStream(HttpStream, ABC):
+    url_base = "https://api.github.com/"
+
+    primary_key = "id"
+
+    # GitHub pagination could be from 1 to 100.
+    page_size = 100
+
+    # These fields will be used for data clearing. Put here keys which represent
+    # objects `{}`, like `user`, `actor` etc.
+    object_fields = ()
+
+    # These fields will be used for data clearing. Put here keys which represent
+    # lists `[]`, like `labels`, `assignees` etc.
+    list_fields = ()
+
+    def __init__(self, repository: str, **kwargs):
+        super().__init__(**kwargs)
+        self.owner, self.repo = repository.split("/")
+        self.repository = repository
+        self._page = 1
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        if response.json():
+            self._page += 1
+            return {"page": self._page}
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        try:
+            yield from super().read_records(
+                sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+            )
+        except HTTPError as e:
+            error_msg = str(e)
+
+            # This whole try/except situation in `read_records()` isn't good but right now in `self._send_request()`
+            # function we have `response.raise_for_status()` so we don't have much choice on how to handle errors.
+            # We added this try/except code because for private repositories `Teams` stream is not available and we get
+            # "404 Client Error: Not Found for url: https://api.github.com/orgs/sherifnada/teams?per_page=100" error.
+            if "/teams?" in error_msg:
+                error_msg = f"Syncing Team stream isn't available for repository {self.repository}"
+
+            self.logger.warn(error_msg)
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+
+        params = {"per_page": self.page_size}
+
+        if next_page_token:
+            params.update(next_page_token)
+
+        return params
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        # Without sending `User-Agent` header we will be getting `403 Client Error: Forbidden for url` error.
+        return {
+            "User-Agent": "PostmanRuntime/7.28.0",
```

Comment:
Is it really necessary?

Reply:
Yes it's really necessary.

Response with `User-Agent` header:

![image](https://user-images.githubusercontent.com/19872253/122520591-77a69680-d01c-11eb-81d4-33b69600c623.png)

Response without `User-Agent` header:

![image](https://user-images.githubusercontent.com/19872253/122520637-868d4900-d01c-11eb-8951-7f59684ce1c6.png)